src/codec/mod.rs

```diff
@@ -37,15 +44,35 @@ impl<T, B> Codec<T, B>
             .length_field_length(3)
             .length_adjustment(9)
             .num_skip(0) // Don't skip the header
-            // TODO: make this configurable and allow it to be changed during
-            // runtime.
-            .max_frame_length(frame::DEFAULT_MAX_FRAME_SIZE as usize)
+            .max_frame_length(max_frame_length)
             .new_read(framed_write);
 
         let inner = FramedRead::new(delimited);
 
         Codec { inner }
     }
+
+    /// Updates the max frame setting.
+    ///
+    /// The change takes effect the next time a frame is decoded. In other
+    /// words, if a frame is currently in process of being decoded with a frame
+    /// size greater than `val` but less than the max frame length in effect
+    /// before calling this function, then the frame will be allowed.
+    #[inline]
+    pub fn set_max_frame_length(&mut self, val: usize) {
```

Comment:
Could this fn be moved to the impl block below (without the trait bounds) and be named `set_max_recv_frame_size` (to match the pattern for sending)?

I used _size as a suffix to match the terminology in the HTTP/2.0 spec.

Reply:
Ah, sure, this naming scheme was based on `FramedRead`  I'll change that.